src/dataDisplay/popover/popover.jsx

```diff
@@ -1,111 +1,278 @@
-import { get } from 'lodash';
-import Classnames from 'classnames';
+import {
+    get,
+    isFunction,
+} from 'lodash';
+import ClassNames from 'classnames';
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
 import PropTypes from 'prop-types';
-import React from 'react';
-import Tooltip from '@material-ui/core/Tooltip';
+import MUIPopper from '@material-ui/core/Popper';
+import React, {
+    useCallback,
+    useEffect,
+    useState,
+} from 'react';
 import {
     UI_CLASS_NAME,
     BEM_POPOVER,
 } from '../../global/constants';
+import Grow from '../../utils/grow';
 import makeStyles from '../../styles/makeStyles';
 
 const propTypes = {
-    /**
-     * The arrow can be enabled/disabled
-     */
-    arrow: PropTypes.bool,
     /**
      * Override or extend the styles applied to Popover.
      */
     classes: PropTypes.shape({
-        root: PropTypes.string,
+        arrow: PropTypes.string,
+        content: PropTypes.string,
+        popoverRoot: PropTypes.string,
         popper: PropTypes.string,
     }),
     /**
      * Additional classes.
      */
     className: PropTypes.string,
     /**
-     * Content
+     * The anchor element.
      */
-    content: PropTypes.oneOfType([
-        PropTypes.string,
-        PropTypes.node,
-    ]),
+    children: PropTypes.node.isRequired,
     /**
-     * We can pass either a string or jsx elements
+     * The content in the Popover.
      */
-    children: PropTypes.shape({}).isRequired,
+    content: PropTypes.node.isRequired,
+    /**
+     * If `true`, Popover will not be fired.
+     */
+    disable: PropTypes.bool,
     /**
      * The `id` of the Popover.
      */
     id: PropTypes.string,
     /**
-     * We can define the max width for the Popover
+     * The max-width of the Popover
      */
-    maxWidth: PropTypes.oneOfType([
-        PropTypes.number,
-        PropTypes.string,
-    ]),
+    maxWidth: PropTypes.number,
+    /**
+     * Popper.js is based on a "plugin-like" architecture,
```

Comment:
Grammar Nitpick: the comma at the end of this line should probably be a semicolon.

Reply:
Ding dang MUI and their grammar.

Reply:
Maybe I should fork them and send them grammar fix PRs... 🤔